eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -429,21 +447,29 @@ private[journal] object ReplicatedCassandra {
                                             .some
                                         }
                                       } yield {
-                                        (deleteTo.value, seqNr)
+                                        // prune yet non-deleted part of journal
+                                        (deleteTo.value, headSeqNr)
                                       }
-                                    } else if (deleteTo.value == seqNr) {
-                                      (seqNr, seqNr).pure[F]
+                                    } else if (deleteTo.value == headSeqNr) {
+                                      // all is fully pruned - nothing to do
+                                      (headSeqNr, headSeqNr).pure[F]
                                     } else {
+                                      // somehow `deleteTo` is ahead of head's `seqNr`
+                                      // TODO MR how this can happen? - in `def append` we clamp down `DeleteTo`
                                       val seqNr1 = deleteTo.value
                                       for {
+                                        // advance head's `seqNr` to catch the `deleteTo` value
                                         _ <- update(seqNr1)
                                         _ <- journalHeadRef.set {
                                           journalHead
                                             .copy(seqNr = seqNr1)
                                             .some
                                         }
                                       } yield {
-                                        (seqNr, seqNr1)
+                                        // assume that journal has been pruned till actual `deleteTo`
+                                        // and `seqNr` was not advanced for some reason
+//                                        (headSeqNr, seqNr1) // TODO MR do we need to prune this part of journal?
+                                        (seqNr1, seqNr1)
```

Comment:
This is an optimization to not issue useless `journal` cleanup query, but I am not sure, if we that is correct.

This part was incompletely covered with unit-tests originally, but if we assume that previous `delete` action was executed fully and correctly, then this should be OK and I can remove the `// TODO` line

Reply:
I do not agree with the change here: if `deleteTo` is ahead of `seqNr` (both in meta) then KJ is in state:
- with inaccessible events in `journal` thus those events should be deleted
- with incorrect record in `metajournal` where `deleteTo > seqNr` thus it should be fixed

Please revert change on the line & cover it with tests (might be you already did)

Reply:
The assumption for this optimization is that previously `Delete` action was completed fully, like we have state `(delete_to: 575, seq_nr: 575)` - thus there is nothing expected in `journal` table (covered by unit-test `test(s"purge meta journal only...`, which doesn't expect any activity on `journal` table (based on assumption that `delete_to` was previously set correctly after corresponding events from `journal` were removed).

If we assume that `delete_to` could have been set wrongly and there are stale events in `journal` table, then on all `prune` actions we have to explicitly issue full cleanup of `journal` table - it is easily possible, though, depending on `segment_size`, can generate few to many queries (more for longer-lived entities).

Reply:
if we go with explicit cleanup, then we have to adjust unit-tests to expect activity on `journal` table too.

Reply:
> thus there is nothing expected in journal table

its wrong assumption because now (probably) some events are left behind and purging such journal will make them orphan. Cleaning orphan events are long and heavy job thus we will not run it often. I see no way how I can approve the PR with such "optimization"